drivers/flash/jesd216.h

```diff
@@ -0,0 +1,416 @@
+/*
+ * Copyright (c) 2020 Peter Bigot Consulting, LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVERS_FLASH_JESD216_H_
+#define ZEPHYR_DRIVERS_FLASH_JESD216_H_
+
+#include <errno.h>
+#include <sys/byteorder.h>
+#include <sys/util.h>
+#include <zephyr/types.h>
+
+/* Following are structures and constants supporting the JEDEC Serial
+ * Flash Discoverable Parameters standard, JESD216 and its successors,
+ * available at
+ * https://www.jedec.org/standards-documents/docs/jesd216b
+ */
+
+#define JESD216_CMD_READ_SFDP   0x5A
+#define JESD216_CMD_BURST_SFDP  0x5B
+
+/* Layout of a JESD216 parameter header. */
+struct jesd216_param_header {
+	uint8_t id_lsb;		/* ID LSB */
+	uint8_t rev_minor;		/* Minor revision number */
+	uint8_t rev_major;		/* Major revision number */
+	uint8_t len_dw;		/* Length of table in 32-bit DWORDs */
+	uint8_t ptp[3];		/* Address of table in SFDP space (LSB@0) */
+	uint8_t id_msb;		/* ID MSB */
+};
```

Comment:
Shouldn't this structure be marked as `__packed`? It is read directly from the flash chip. right? Similarly `jesd216_sfdp_header`.

Reply:
It's not necessary; both of them are naturally aligned.  It's also not desirable: making `jesd216_sfdp_header` would allow a variable declaration to be placed at an address where word alignment isn't guaranteed, thus increasing the code size required to cope with misaligned access to the `uint32_t` magic field.

Reply:
They are naturally aligned, but compilers are free to insert any padding between the members of structures. It is only guaranteed that there will be no padding before the first member. Of course, it does not seem reasonable to use any padding in those two structures in question, but there are no guarantees in this regard.

Reply:
OK; as I read ISO/IEC 9899:2017 section 6.7.2.1#15 it is indeed possible that a compiler or ABI might introduce padding between these fields.   I think this is very unlikely to happen due to quality of implementation concerns (people wouldn't use a compiler that wasted space with these declarations).

But experimenting it doesn't seem to hurt.  Done.